src/components/image/_image.scss

```
@@ -11,21 +11,58 @@
   min-height: 1px; /* 1 */
   line-height: 0; // Fixes cropping when image is resized by forcing its height to be determined by the image not line-height
 
+  .euiImage__img {
+    width: 100%;
+  }
+
   &.euiImage--hasShadow {
     .euiImage__img {
       @include euiBottomShadowMedium;
     }
   }
 
-  &.euiImage--allowFullScreen:hover {
-    .euiImage__img {
-      cursor: pointer;
+  &.euiImage--allowFullScreen {
+    .euiImage__button {
+      position: absolute;
+      display: block;
+      left: 0;
+      top: 0;
+      width: 100%;
+      height: 100%;
     }
 
     .euiImage__icon {
+      // transition the shadow
+      transition: all $euiAnimSpeedFast $euiAnimSlightResistance;
```

There is no shadow on the icon, I think the comment is supposed to be opacity in which the property here should change to that.
```suggestion
      transition: opacity $euiAnimSpeedFast $euiAnimSlightResistance;
```

---

I found out the issue was because the icon is actually an SVG and doesn't support `opacity`. So I changed to `fill-opacity` and it's working.